app/models/rbac.rb

```diff
@@ -240,34 +243,28 @@ def self.get_managed_filter_object_ids(klass, scope, filter)
     scope.find_tags_by_grouping(filter, :ns => '*', :select => minimum_columns_for(klass)).reorder(nil).collect(&:id)
   end
 
-  def self.find_targets_with_direct_rbac(scope, rbac_filters, find_options = {}, user_or_group = nil)
-    filtered_ids, u_filtered_ids = calc_filtered_ids(scope, rbac_filters, user_or_group)
+  def self.find_targets_with_direct_rbac(scope, rbac_filters, find_options, user, miq_group)
+    filtered_ids, u_filtered_ids = calc_filtered_ids(scope, rbac_filters, user, miq_group)
     find_targets_filtered_by_ids(scope, find_options, u_filtered_ids, filtered_ids)
   end
 
-  def self.find_targets_with_user_group_rbac(klass, scope, _rbac_filters, find_options = {}, user_or_group = nil)
-    if user_or_group && user_or_group.self_service?
```

Comment:
@gtanzillo good question:

That condition was moved from `find_targets_with_user_group_rbac` to `apply_user_group_rbac_to_class?` for 2 reasons:

1. The other blocks in `find_targets_with_rbac` all did a "should I perform this rbac?" and then "perform it"
2. If this check failed, it would run without rbac, which was essentially just the logic built into `find_targets_with_rbac`

I put that into a single commit [Rbac.find_targets_with_user_group_rbac extract condition]((https://github.com/ManageIQ/manageiq/pull/7141/commits/caa24fb4b514630c94aa8c2703dc006768b6e33c)